keras_cv/layers/preprocessing/random_flip.py

```diff
@@ -0,0 +1,201 @@
+                "RandomFlip layer {name} received an unknown mode="
+                "{arg}".format(name=self.name, arg=mode)
+            )
+        self.auto_vectorize = True
+        self.bounding_box_format = bounding_box_format
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def augment_image(self, image, transformation, **kwargs):
+        flipped_output = image
+
+        def flip_horizontal():
+            return tf.image.flip_left_right(flipped_output)
+
+        flipped_output = tf.cond(
+            tf.cast(transformation["flip_horizontal"], dtype=tf.bool),
+            flip_horizontal,
+            lambda: flipped_output,
+        )
+
+        def flip_vertical():
+            return tf.image.flip_up_down(flipped_output)
+
+        flipped_output = tf.cond(
+            tf.cast(transformation["flip_vertical"], dtype=tf.bool),
+            flip_vertical,
+            lambda: flipped_output,
+        )
+        flipped_output.set_shape(image.shape)
+        return flipped_output
+
+    def get_random_transformation(self, **kwargs):
+        flip_horizontal = False
+        flip_vertical = False
+        if self.horizontal:
+            flip_horizontal = self._random_generator.random_uniform(shape=[]) > 0.5
+        if self.vertical:
+            flip_vertical = self._random_generator.random_uniform(shape=[]) > 0.5
+        return {
+            "flip_horizontal": flip_horizontal,
+            "flip_vertical": flip_vertical,
+        }
+
+    def augment_bounding_boxes(
+        self, bounding_boxes, transformation=None, image=None, **kwargs
+    ):
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomFlip()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomFlip(bounding_box_format='xyxy')`"
+            )
+
+        bounding_boxes = bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            images=image,
+        )
+        image_shape = tf.shape(image)
+        h = tf.cast(image_shape[H_AXIS], dtype="float32")
+        w = tf.cast(image_shape[W_AXIS], dtype="float32")
+        bounding_boxes_out = tf.identity(bounding_boxes)
+
+        def flip_horizontal():
+            return tf.stack(
+                [
+                    w - bounding_boxes_out[:, 2],
+                    bounding_boxes_out[:, 1],
+                    w - bounding_boxes_out[:, 0],
+                    bounding_boxes_out[:, 3],
+                ],
+                axis=-1,
+            )
+
+        bounding_boxes_out = tf.cond(
+            tf.cast(transformation["flip_horizontal"], dtype=tf.bool),
+            flip_horizontal,
+            lambda: bounding_boxes_out,
+        )
+
+        def flip_vertical():
```

Comment:
Can we define some helper methods instead of inlining all of these:

```
def _flip_bounding_boxes_vertical(boxes, image_height):
            return tf.stack(
                [
                    boxes[:, 0],
                    image_height - boxes[:, 3],
                    boxes[:, 2],
                    image_height - boxes[:, 1],
                ],
                axis=-1,
            )
```

then in the tf.cond do:

```
bounding_boxes = tf.cond(transformation["flip_vertical"], lambda: _flip_bounding_boxes_vertical(bounding_boxes), lambda: bounding_boxes)
bounding_boxes = tf.cond(transformation["flip_horizontal"], lambda: _flip_bounding_boxes_horizontal(bounding_boxes), lambda: bounding_boxes)
```

which is more readable, and removes the need to define these inline functions.

Additionally, does using `rel_xyxy` remove the need for a height parameter here?  If you use rel_xyxy I think the flip function will just work with `1 - bounding_boxes_out[:, 3]`.

Reply:
yes, that would reduce these three lines -        
 image_shape = tf.shape(image)
  h = tf.cast(image_shape[H_AXIS], dtype="float32")
  w = tf.cast(image_shape[W_AXIS], dtype="float32")

Reply:
Done!